aidants_connect_pico_cms/models.py

```diff
@@ -0,0 +1,36 @@
+from django.db import models
+from django.utils.html import mark_safe
+
+from markdown import markdown
+
+from aidants_connect_web.models import Aidant
+
+
+class CmsContent(models.Model):
+    created_at = models.DateTimeField("Date de création", auto_now_add=True)
+    updated_at = models.DateTimeField("Date de modification", auto_now=True)
+    updated_by = models.ForeignKey(
+        Aidant, models.SET_NULL, blank=True, null=True, verbose_name="Modifié par"
+    )
+    published = models.BooleanField("Publié")
+    slug = models.SlugField("Clé d’URL")
+    sort_order = models.PositiveSmallIntegerField("Tri", null=True, db_index=True)
+    body = models.TextField("Contenu")
+
+    def to_html(self):
+        return mark_safe(markdown(self.body))
```

Comment:
Est-ce que c'est safe de marquer ceci comme safe ? C'est du contenu BDD.

Reply:
Ça dépend de si la fonction `markdown` échappe le HTML contenu dans le markdown. Sinon, ça veut dire qu'il est possible de faire ça :

```markdown
# Mon témoignage legit

<script>
  alert("coucou")
</script>
```

Reply:
Alors. Le JS se retrouve tel quel dans le code HTML.
MAIS. Notre CSP empêche l'exécution de code inline.
Un avis ?

![Capture d’écran 2023-01-09 à 16 19 49](https://user-images.githubusercontent.com/1035145/211342798-8ecf2382-f66d-49d5-9c4b-e363da68f494.png)

Reply:
Alors il est vrai que le CSP normalement bloque toutes les executions inline. Donc on est à priori safe.

après est ce faire un force_escape sur self.body ) ca irait pas ?

Reply:
Le problème avec force_escape c'est que ça donne ça : 

![Capture d’écran 2023-01-09 à 17 29 31](https://user-images.githubusercontent.com/1035145/211358211-b643ed39-6335-4671-bd36-db02441ab8eb.png)

Reply:
Chuis allé voir le code de `force_escape`, par curiosité. Il fait déjà un `mark_safe`. Tu coup, tu peux te contenter de : 
```suggestion
        return force_escape(markdown(self.body))
```

Reply:
C'est pas ça qu'on veut faire en fait, parce que en sortie du markdown on a plein de HTML légitime qu'on ne veut pas échapper. Par contre on peut échapper le self.body avant de le mettre dans la fonction markdown, et en effet ça marche. C'est ce que suggérait JM mais j'avais pas compris tout de suite.